File: content/search/source/includes/view/code-snippets/csharp/FacetFieldsQuery.cs
using MongoDB.Bson;
using MongoDB.Driver;

// Connect to your Atlas deployment
var uri = "<connectionString>";
var client = new MongoClient(uri);
var database = client.GetDatabase("sample_airbnb");
var collection = database.GetCollection<BsonDocument>(
    "listings_SearchableTypes");

// Run the aggregation query on the View
var pipeline = new[]
{
    new BsonDocument("$searchMeta",
        new BsonDocument("index", "listingsSearchableTypes")
            .Add("facet",
                new BsonDocument("operator",
                    new BsonDocument("text",
                        new BsonDocument("path", "summary")
                            .Add("query", "ocean view")))
                    .Add("facets",
                        new BsonDocument("idFacet",
                            new BsonDocument("type", "string")
                                .Add("path", "idString")
                                .Add("numBuckets", 10))
                            .Add("hostFacet",
                                new BsonDocument("type", "string")
                                    .Add("path",
                                        "superHostString")))))
};

var results = collection.Aggregate<BsonDocument>(pipeline);
foreach (var doc in results.ToList())
{
    Console.WriteLine(doc);
}

File: content/search/source/includes/view/code-snippets/nodejs/facet-fields-query.js
import { MongoClient } from "mongodb";

// Connect to your Atlas deployment
const uri = "<connectionString>";
const client = new MongoClient(uri);

async function run() {
  try {
    const database = client.db("sample_airbnb");
    const collection = database.collection(
      "listings_SearchableTypes");

    // Run the aggregation query on the View
    const pipeline = [
      {
        $searchMeta: {
          index: "listingsSearchableTypes",
          facet: {
            operator: {
              text: {
                path: "summary",
                query: "ocean view"
              }
            },
            facets: {
              idFacet: {
                type: "string",
                path: "idString",
                numBuckets: 10
              },
              hostFacet: {
                type: "string",
                path: "superHostString"
              }
            }
          }
        }
      }
    ];

    const results = await collection.aggregate(pipeline).toArray();
    console.log(results);
  } finally {
    await client.close();
  }
}
run().catch(console.dir);
